python/lsst/cp/pipe/ptc.py

```diff
@@ -0,0 +1,535 @@
+    )
+    isrMandatorySteps = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that must be performed for valid results. Raises if any of these are False.",
+        default=['doAssembleCcd']
+    )
+    isrForbiddenSteps = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that must NOT be performed for valid results. Raises if any of these are True",
+        default=['doFlat', 'doFringe', 'doAddDistortionModel', 'doBrighterFatter', 'doUseOpticsTransmission',
+                 'doUseFilterTransmission', 'doUseSensorTransmission', 'doUseAtmosphereTransmission']
+    )
+    isrDesirableSteps = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that it is advisable to perform, but are not mission-critical." +
+        " WARNs are logged for any of these found to be False.",
+        default=['doBias', 'doDark', 'doCrosstalk', 'doDefect']
+    )
+    isrUndesirableSteps = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that it is *not* advisable to perform in the general case, but are not" +
+        " forbidden as some use-cases might warrant them." +
+        " WARNs are logged for any of these found to be True.",
+        default=['doLinearize']
+    )
+    ccdKey = pexConfig.Field(
+        dtype=str,
+        doc="The key by which to pull a detector from a dataId, e.g. 'ccd' or 'detector'.",
+        default='ccd',
+    )
+    makePlots = pexConfig.Field(
+        dtype=bool,
+        doc="Plot the PTC curves?.",
+        default=False,
+    )
+    ptcFitType = pexConfig.ChoiceField(
+        dtype=str,
+        doc="Fit PTC to approximation in Astier+19 (Equation 16) or to a polynomial.",
+        default="POLYNOMIAL",
+        allowed={
+            "POLYNOMIAL": "n-degree polynomial (use 'polynomialFitDegree' to set 'n').",
+            "ASTIERAPPROXIMATION": "Approximation in Astier+19 (Eq. 16)."
+        }
+    )
+    polynomialFitDegree = pexConfig.Field(
+        dtype=int,
+        doc="Degree of polynomial to fit the PTC, when 'ptcFitType'=POLYNOMIAL.",
+        default=2,
+    )
+    binSize = pexConfig.Field(
+        dtype=int,
+        doc="Bin the image by this factor in both dimensions.",
+        default=1,
+    )
+    minMeanSignal = pexConfig.Field(
+        dtype=float,
+        doc="Minimum value of mean signal (in ADU) to consider.",
+        default=0,
+    )
+    maxMeanSignal = pexConfig.Field(
+        dtype=float,
+        doc="Maximum value to of mean signal (in ADU) to consider.",
+        default=9e6,
+    )
+    sigmaCutPtcOutliers = pexConfig.Field(
+        dtype=float,
+        doc="Sigma cut for outlier rejection in PTC",
+        default=5.0,
+    )
+    maxIterationsPtcOutliers = pexConfig.Field(
+        dtype=int,
+        doc="Maximum number of iterations for outlier rejection in PTC",
+        default=5,
+    )
+
+
+class MeasurePhotonTransferCurveTask(pipeBase.CmdLineTask):
```

Comment:
I'm ignoring the `pipelineTask` and butler gen3 conversion work that will need to be done, as we don't yet have a formal plan.  I've interspersed some of my ideas below on how that conversion might be done.